pkg/reconciler/reconciler.go

```diff
@@ -100,13 +103,24 @@ func NewBase(ctx context.Context, controllerAgentName string, cmw configmap.Watc
 		}()
 	}
 
+	statsReporter := GetStatsReporter(ctx)
+	if statsReporter == nil {
+		logger.Debug("Creating stats reporter")
+		var err error
+		statsReporter, err = NewStatsReporter(controllerAgentName)
+		if err != nil {
+			logger.Fatal(err)
+		}
+	}
+
 	base := &Base{
```

Comment:
Could you try switching this to use NewBase constructor function, and move initializing the StatsReporter to that function.